src/lib/rmt/maria_db/current_root_password_dialog.rb

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2018 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require 'rmt/base'
+
+module RMT; end
+module RMT::MariaDB; end
+
+class RMT::MariaDB::CurrentRootPasswordDialog < RMT::Base
```

Comment:
to shorted main method I suggest to use as base Dialog base class - https://github.com/yast/yast-yast2/blob/master/library/general/src/lib/ui/dialog.rb
BTW this RMT::Base looks more like mixin then base class.

Reply:
`Dialog` looks nice :+1: 
There's quite a lot of mixing in happening in Yast, a simple base class is easier to handle by an untrained mind :grimacing: